### DISCUSSION TITLE
Adjust rendering of bars by one character

### DISCUSSION THREAD

**Discussion by petern3** [no-chris/chord-mark#533] [Q&A] [answered]
When reading the rendered song, where the renderer places the chords can be a little misleading. For example:

```
C.. Am..
I _heard there was a _secret chord
```
Becomes
```
  |C                Am          |
I heard there was a secret chord
```

Note that the Am is in the correct place, but the C is bumped a little because of the vertical bar, making it look like that chord is played slightly later than is defined. Could this be adjusted?

**Comment by no-chris** [comment]
Yes, the current rendering is the result of an intense debate with myself on what the rendering should look like. What you are suggesting is instead of having this:

```
  |C                Am          |
I heard there was a secret chord
```
to have this:
```
  |C                Am          |
I  heard there was a secret chord
```

I agree it can be a bit misleading at first but I decided on the current format because:
- it helps keeping lines short (which avoid line breaks when printing, especially in multiple columns)
- I find that adding spaces in lyrics makes them harder to read so I'd rather avoid this as much as possible
- I also think it's more readable when the lyrics starts on the first beat of the first bar

```
|C                G   
Barefoot on the grass
|C                G   
 Barefoot on the grass
```

I found that after a little while of reading chord charts, the inconvenience of having the slight misalignment disappears. 

Of course, happy to discuss this further if you disagree

**Reply by petern3** [reply]
If it's an easy change, perhaps it could be a setting?

* _"it helps keeping lines short (which avoid line breaks when printing, especially in multiple columns)"_ - It wouldn't change all lines, but only some would increase by 1 char (~2% longer). (See also the next comment). That's less than the extra spaces added at the start of a line if the chord begins _before_ the lyrics
* _"I find that adding spaces in lyrics makes them harder to read so I'd rather avoid this as much as possible"_ - Some cases will be worse, some will be better

```
Lyric example: L_o_ong word

Current behaviour -> Proposed behaviour

 |A |B |        |A |B |
Lo  ong word -> Lo  ong word  (The same)

 |A B |         |A B |
Lo  ong word -> Lo ong word  (Lyrics have less spacing)

|B  A |B  A|      |B  A |B   A|
   Lo ong word ->    Lo  ong word  (Lyrics have more spacing)
```

* _"I also think it's more readable when the lyrics starts on the first beat of the first bar"_ - I don't think so, but that's subjective and not a particularly strong opinion.

I personally think being easier to read is more important than having one fewer characters on some lines, particularly if someone unfamiliar with the format is reading off it. That said, don't read my long response as "I strongly think this should be changed", but "If it's not much effort I think it would be better, but I could get used to it as-is" 🙂

**Reply by no-chris** [reply]
> It it's an easy change, perhaps it could be a setting?

Yeah, I somehow forgot to mention that the chosen option was also the easiest one to implement, and by far 😅 

But I don't have a strong opinion either, and all the points you raise are very valid. Probably most people would agree with you, also. I think it's too early to make a definitive decision either way. When I have a bit of time on my hands I can experiment with trying to implement your suggested way. In the meantime we can leave the topic open for discussion

**Reply by petern3** [reply]
Sounds good!

**Comment by no-chris** [comment, marked as answer]
Hello!

Happy 2023.

This has been implemented in 0.13.0, for now available as a beta: https://github.com/no-chris/chord-mark/releases/tag/v0.13.0-beta.0
Beta testing, if you feel like it, is possible here: https://deploy-preview-610--chord-chart-studio.netlify.app/

Cheers!

**Reply by no-chris** [reply]
related PR: https://github.com/no-chris/chord-mark/pull/592
